### PR TITLE
restore k8s patch versions

### PR DESCRIPTION
**Pull request by jackfrancis**
**What this PR does / why we need it**: Restores referential integrity in acs-engine of prior working patch versions of Kubernetes. Also rationalized duplicate version constants (moved duplicates to `pkg/api/common`) and rationalized existing validations.

**Release note**:

```
NONE
```

### CODE REVIEW

File: pkg/acsengine/k8s_versions.go [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+		"dashboard":       "kubernetes-dashboard-amd64:v1.6.3",
+		"exechealthz":     "exechealthz-amd64:1.2",
+		"addonresizer":    "addon-resizer:1.7",
+		"heapster":        "heapster-amd64:v1.4.1",
+		"dns":             "k8s-dns-kube-dns-amd64:1.14.4",
+		"addonmanager":    "kube-addon-manager-amd64:v6.4-beta.2",
+		"dnsmasq":         "k8s-dns-dnsmasq-nanny-amd64:1.14.4",
+		"pause":           "pause-amd64:3.0",
+		"tiller":          DefaultTillerImage,
+		"windowszip":      "v1.7.2intwinnat.zip",
```
Comment (jackfrancis) [inline pkg/acsengine/k8s_versions.go]: // TODO we never built a windows v1.7.4 artifact @JiangtianLi

**Comment by JiangtianLi** [comment]
@jackfrancis  on it.

**Comment by JiangtianLi** [comment]
@jackfrancis  1.7.4 built and uploaded.

**Comment by jackfrancis** [comment]
@JiangtianLi thanks! is the artifact name `v1.7.4-1intwinnat.zip` ?

**Comment by jackfrancis** [comment]
@amanohar @dmitsh @JackQuincy @weinong  One of you should take a look over the exact implementation of this patch version PR. Thanks!

**Comment by JiangtianLi** [comment]
@jackfrancis yes

**Pull request merged**